Sources/PortalSwift/Yield/YieldXyz.swift

```diff
@@ -0,0 +1,88 @@
+//
+//  YieldXyz.swift
+//  PortalSwift
+//
+//  Created by Ahmed Ragab on 20/10/2025.
+//
+
+import Foundation
+
+/// YieldXyz provider implementation for discovering and managing yield opportunities.
+public class YieldXyz {
+    private let api: PortalYieldXyzApi
+    
+    /// Create an instance of YieldXyz.
+    /// - Parameter api: The PortalYieldXyzApi instance to use for yield operations.
+    public init(api: PortalYieldXyzApi) {
+        self.api = api
+    }
+    
+    /// Discovers yield opportunities based on the provided parameters.
+    /// - Parameter request: Optional parameters for yield discovery. If nil, uses default parameters.
+    /// - Returns: A `GetYieldsXyzResponse` containing available yield opportunities.
+    /// - Throws: An error if the operation fails.
+    public func discover(request: GetYieldsXyzRequest? = nil) async throws -> GetYieldsXyzResponse {
+        let discoveryRequest = request ?? GetYieldsXyzRequest()
+        return try await api.getYields(request: discoveryRequest)
+    }
+    
+    /// Enters a yield opportunity with the specified parameters.
+    /// - Parameter request: The parameters for entering a yield opportunity.
+    /// - Returns: An `EnterYieldXyzResponse` containing the action details.
+    /// - Throws: An error if the operation fails.
+    public func enter(request: EnterYieldXyzRequest) async throws -> EnterYieldXyzResponse {
+        return try await api.enterYield(request: request)
+    }
+    
+    /// Tracks a transaction by submitting its hash to the Yield.xyz integration.
+    /// - Parameters:
+    ///   - transactionId: The ID of the transaction to track.
+    ///   - txHash: The hash of the transaction to submit.
+    /// - Returns: `true` if the submission was successful.
+    /// - Throws: An error if the operation fails.
+    @discardableResult
+    public func track(transactionId: String, txHash: String) async throws -> Bool {
```

Comment:
Might need to update the returned response here if we surface the API endpoint response instead of a simple bool